ray-curator/tests/stages/text/embedders/test_utils.py

```diff
@@ -0,0 +1,45 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import cudf
+import cupy as cp
+
+from ray_curator.stages.text.embedders.utils import create_list_series_from_1d_or_2d_ar
+
+
+class TestCreateListSeriesFrom1dOr2dAr:
+    """Test create_list_series_from_1d_or_2d_ar function."""
+
+    def test_embedding_creation_like_base_usage(self):
+        """Test the function as used in EmbeddingModelStage.create_output_dataframe."""
+        # Simulate embeddings from a model (2D array: batch_size x embedding_dim)
+        collected_output = cp.array([[0.1, 0.2, 0.3], [0.4, 0.5, 0.6], [0.7, 0.8, 0.9]])
+
+        # Create a GPU dataframe with index like in the actual usage
+        df_gpu = cudf.DataFrame(index=cudf.RangeIndex(3))
```

Comment:
Could you test with this Reordered/shuffled please. 
```suggestion
        df_gpu = cudf.DataFrame(index=cudf.RangeIndex(3))
        df_gpu = df_gpu.iloc[[2, 0, 1]]
```